src/main/java/rx/observers/TestSubscriber2.java

```diff
@@ -18,184 +18,70 @@
-        return ts.getCompletions();
-    }
-
-    @Override
-    public void onError(Throwable e) {
-        ts.onError(e);
-    }
-
-    public List<Throwable> getOnErrorEvents() {
-        return ts.getOnErrorEvents();
-    }
-
-    @Override
-    public void onNext(T t) {
-        ts.onNext(t);
-    }
-
-    public final int getValueCount() {
-        return ts.getValueCount();
-    }
-
-    public TestSubscriber2<T> requestMore(long n) {
-        ts.requestMore(n);
-        return this;
-    }
-
-    public List<T> getOnNextEvents() {
-        return ts.getOnNextEvents();
-    }
-
-    public TestSubscriber2<T> assertReceivedOnNext(List<T> items) {
-        ts.assertReceivedOnNext(items);
-        return this;
-    }
-
-    public final boolean awaitValueCount(int expected, long timeout, TimeUnit unit) {
-        return ts.awaitValueCount(expected, timeout, unit);
-    }
-
-    public TestSubscriber2<T> assertTerminalEvent() {
-        ts.assertTerminalEvent();
-        return this;
-    }
-
-    public TestSubscriber2<T> assertUnsubscribed() {
-        ts.assertUnsubscribed();
-        return this;
-    }
-
-    public TestSubscriber2<T> assertNoErrors() {
-        ts.assertNoErrors();
-        return this;
-    }
-
-    public TestSubscriber2<T> awaitTerminalEvent() {
-        ts.awaitTerminalEvent();
-        return this;
-    }
-
-    public TestSubscriber2<T> awaitTerminalEvent(long timeout, TimeUnit unit) {
-        ts.awaitTerminalEvent(timeout, unit);
-        return this;
-    }
-
-    public TestSubscriber2<T> awaitTerminalEventAndUnsubscribeOnTimeout(long timeout,
-            TimeUnit unit) {
-        ts.awaitTerminalEventAndUnsubscribeOnTimeout(timeout, unit);
-        return this;
-    }
-
-    public Thread getLastSeenThread() {
-        return ts.getLastSeenThread();
-    }
-
-    public TestSubscriber2<T> assertCompleted() {
-        ts.assertCompleted();
-        return this;
-    }
-
-    public TestSubscriber2<T> assertNotCompleted() {
-        ts.assertNotCompleted();
-        return this;
-    }
-
-    public TestSubscriber2<T> assertError(Class<? extends Throwable> clazz) {
-        ts.assertError(clazz);
-        return this;
-    }
-
-    public TestSubscriber2<T> assertError(Throwable throwable) {
-        ts.assertError(throwable);
-        return this;
-    }
-
-    public TestSubscriber2<T> assertNoTerminalEvent() {
-        ts.assertNoTerminalEvent();
-        return this;
-    }
-
-    public TestSubscriber2<T> assertNoValues() {
-        ts.assertNoValues();
-        return this;
-    }
-
-    public TestSubscriber2<T> assertValueCount(int count) {
-        ts.assertValueCount(count);
-        return this;
-    }
-
-    public TestSubscriber2<T> assertValues(T... values) {
-        ts.assertValues(values);
-        return this;
-    }
-
-    public TestSubscriber2<T> assertValue(T value) {
-        ts.assertValue(value);
-        return this;
-    }
+public interface TestSubscriber2<T> extends Observer<T> {
```

Comment:
I'd call this `TestConsumer` because that `2` at the end will make people wonder and question endlessly.